src/poetry/puzzle/provider.py

```diff
@@ -497,16 +497,14 @@ def complete_package(
                 extra_dependencies = package.extras.get(extra, [])
                 for extra_dependency in extra_dependencies:
                     if extra_dependency.name == dependency.name:
-                        stack += list(extra_dependency.extras)
+                        stack += sorted(extra_dependency.extras)
                     else:
                         optional_dependencies.add(extra_dependency.name)
 
             # If some extras/features were required, we need to add a special dependency
             # representing the base package to the current package.
 
-            dependency_package = dependency_package.with_features(
-                list(dependency.extras)
-            )
+            dependency_package = dependency_package.with_features(dependency.extras)
```

Comment:
I guess no need for sorted/list here since it doesn't change the outcome of the method?